py-pkg/pkgnet/dependency_reporter.py

```diff
@@ -0,0 +1,63 @@
+from typing import Optional, Callable
+
+try:
+    import importlib.resources as importlib_resources
+except ImportError:
+    # Python 3.6 uses importlib_resources backport
+    import importlib_resources
+
+
+import pandas as pd
+import pkg_resources
+
+from pkgnet.abstract_graph_reporter import AbstractGraphReporter
+from pkgnet.abstract_package_reporter import registrar
+from pkgnet.graphs import DirectedGraph
+from pkgnet.search_utils import recursive_node_search
+
+
+@registrar.register_reporter
+class DependencyReporter(AbstractGraphReporter):
+
+    _graph_class = DirectedGraph
+
+    report_template = "tab_dependency_report.jinja"
+    report_slug = "dependency-report"
+    report_name = "Dependencies"
+    layout = "kamada_kawai_layout"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._ignore_packages = []
+
+    ### PUBLIC METHODS ###
+
+    @classmethod
+    def report_template(cls) -> (str, str, Optional[Callable]):
+        source = importlib_resources.read_text("pkgnet.templates", "tab_dependency_report.jinja")
+        path = next(importlib_resources.path("pkgnet.templates", "tab_dependency_report.jinja").gen)
+        mtime = path.stat().st_mtime  # last modified time
```

Comment:
could this be called `modified_time` or something more explicit? `mtime` was confusing to me.

Also how does this + the `lambda` below work? Is that value computed right on this line getting stored in that lambda somehow? Or are we relying on weird "found it in a higher environment" stuff? Not saying I'm against it, I'm just really confused about how this is working.

Reply:
I'm open to more clear naming, but this is what `jinja2` docs say to do. https://jinja.palletsprojects.com/en/2.11.x/api/#jinja2.BaseLoader